test/recipes/20-test_cli_list.t

```diff
@@ -14,12 +14,34 @@ use OpenSSL::Test::Utils;
 
 setup("test_cli_list");
 
-plan tests => 2;
-
-ok(run(app(["openssl", "list", "-skey-managers"],
-        stdout => "listout.txt")),
-"List skey managers - default configuration");
-open DATA, "listout.txt";
-my @match = grep /secret key/, <DATA>;
-close DATA;
-ok(scalar @match > 1 ? 1 : 0, "Several skey managers are listed - default configuration");
+plan tests => 3;
+
+sub check_skey_manager_list {
+    my $provider = $_[0];
+    ok(run(app(["openssl", "list", "-skey-managers"],
+               stdout => "listout.txt")),
+       "List skey managers - $provider provider");
+    open DATA, "listout.txt";
+    my @match = grep /secret key/, <DATA>;
+    close DATA;
+    ok(scalar @match > 1 ? 1 : 0,
+       "Several skey managers are listed - $provider provider");
+}
+
+check_skey_manager_list("default");
+
+my $no_fips = disabled('fips') || ($ENV{NO_FIPS} // 0);
+
+subtest fips => sub {
+    plan skip_all => "FIPS is disabled"
+        if $no_fips;
+    plan tests => 2;
```

Comment:
There needs to be a FIPS version check since we guarantee cross version compatibility for the FIPS provider.  I'll kick the CIs and you'll see the failures.